packages/lodestar/src/network/gossip/validation/attestation.ts

```diff
@@ -0,0 +1,131 @@
+      "Rejected gossip committee attestation",
+      {reason: "attestation block is invalid", ...attestationLogContext}
+    );
+    return ExtendedValidatorResult.reject;
+  }
+
+  if(!hasValidAttestationSlot(config, chain.getGenesisTime(), attestation)) {
+    logger.warn("Ignored gossip committee attestation", {reason: "Invalid slot time", ...attestationLogContext});
+    //attestation might be valid later so passing to attestation pool
+    await chain.receiveAttestation(attestation);
+    return ExtendedValidatorResult.ignore;
+  }
+
+  //no other validator attestation for same target epoch has been seen
+  if(await db.seenAttestationCache.hasCommitteeAttestation(attestation)) {
+    return ExtendedValidatorResult.ignore;
+  }
+
+  if(!isUnaggregatedAttestation(attestation)) {
+    logger.warn("Rejected gossip committee attestation", {reason: "not unaggregated", ...attestationLogContext});
+    return ExtendedValidatorResult.reject;
+  }
+
+  const attestationStateContext = await getBlockStateContext(chain.forkChoice, db, attestation.data.beaconBlockRoot);
+  if(!attestationStateContext) {
+    logger.warn(
+      "Ignored gossip committee attestation",
+      {reason: "missing attestation state/block", ...attestationLogContext}
+    );
+    //attestation might be valid after we receive block
+    await chain.receiveAttestation(attestation);
+    return ExtendedValidatorResult.ignore;
+  }
+  const attestationPreStateContext =
+      await getAttestationPreState(config, chain, db, attestation.data.target);
+  if(!attestationPreStateContext) {
+    logger.warn(
+      "Ignored gossip committee attestation",
+      {reason: "missing attestation prestate", ...attestationLogContext}
+    );
+    //attestation might be valid after we receive block
+    await chain.receiveAttestation(attestation);
+    return ExtendedValidatorResult.ignore;
+  }
+
+  const expectedSubnet = computeSubnetForAttestation(config, attestationPreStateContext.epochCtx, attestation);
+  if(subnet !== expectedSubnet) {
+    logger.warn(
+      "Rejected gossip committee attestation",
+      {reason: "wrong subnet", subnet, expectedSubnet, ...attestationLogContext}
+    );
+    return ExtendedValidatorResult.reject;
+  }
+  if(!isValidIndexedAttestation(
+    attestationPreStateContext.epochCtx,
+    attestationPreStateContext.state,
+    attestationPreStateContext.epochCtx.getIndexedAttestation(attestation),
+    true
+  )) {
+    logger.warn(
+      "Rejected gossip committee attestation",
+      {reason: "invalid indexed attestation", ...attestationLogContext}
+    );
+    return ExtendedValidatorResult.reject;
+  }
+  await db.seenAttestationCache.addCommitteeAttestation(attestation);
+  logger.profile("gossipAttestationValidation");
+  logger.info("Received valid committee attestation", attestationLogContext);
+  return ExtendedValidatorResult.accept;
+}
+
+export async function isAttestingToInValidBlock(db: IBeaconDb, attestation: Attestation): Promise<boolean> {
```

Comment:
Should we be checking for block in the block db? Eg: checking for included good block, rather than bad block.

Reply:
This is gossip validation so if we don't have valid block we would put it in pool and ignore so it makes sense to first check if it's invalid as we need to reject immediately